File: C/ex1/stochasiticGradientDescent.c
int stochasiticGradientDescent(int M, double alpha,  double X[], double Y[], double theta[])
{
	int i;
	double temp0, temp1;

		theta[0]  = 0;
    		theta[1]  = 0;
    
    	for(i=0; i < M; i++){
        	theta[0] = theta[0] - alpha*(theta[0] + theta[1] * X[i] - Y[i]) * 1;
        	theta[1] = theta[1] - alpha*(theta[0] + theta[1] * X[i] - Y[i]) * X[i];
    	}
		
	
	return 0;
}
		

File: Fortran/ex1/stochasiticGradientDescent.f90
      subroutine stochasiticGradientDescent(M, alpha, iterations, X, y, theta)
	implicit none
	
	integer:: i, M
	real,dimension(2):: theta
	real,dimension(M):: X, y
		
	
	      theta(1)  = 0
    	      theta(2)  = 0   
    	do i = 1, M
        	theta(1) = theta(1) - (alpha)*(theta(1) + theta(2) * X(i) - y(i)) * 1.
        	theta(2) = theta(2) - (alpha)*(theta(1) + theta(2) * X(i) - y(i)) * X(i);
    	enddo
		
		
	
	end subroutine stochasiticGradientDescent
